scripts/deployment.ts

```diff
@@ -1,32 +1,62 @@
 import { ethers } from 'hardhat'
 import { ContractFactory, Contract } from 'ethers'
 import '@nomiclabs/hardhat-ethers'
+import { run } from 'hardhat'
 
-interface RollupCreatedEvent {
-  event: string;
-  address: string;
-  args?: {
-    rollupAddress: string;
-    inboxAddress: string;
-    adminProxy: string;
-    sequencerInbox: string;
-    bridge: string;
-  };
-}
+// Define a verification function
+async function verifyContract(
+  contractName: string,
+  contractAddress: string,
+  constructorArguments: any[] = [],
+  contractPathAndName?: string // optional
+): Promise<void> {
+  try {
+    // Define the verification options with possible 'contract' property
+    const verificationOptions: {
+      contract?: string
+      address: string
+      constructorArguments: any[]
+    } = {
+      address: contractAddress,
+      constructorArguments: constructorArguments,
```

Comment:
Let's also specify the contract to verify here, I think it is
```suggestion
      contract: contractName,
      address: contractAddress,
      constructorArguments: constructorArguments,
```
but please test

Reply:
We don't need it for all contracts Harry. The only contract that had to different codes with same bytecode in the repo, was bridge contract.
I put contract as optional here, because otherwise we need to put the path for all other contracts which makes the code unclean and also we don't need it, cause for other contracts, hardhat was able to find the right contract just by name.

Reply:
wdym? This is just passing the contract name right? The current setup is a little but unsafe since say you pass in 0xabcd and Inbox, it will verify and output "Inbox contract verified" even if 0xabcd is Outbox because hardhat will just find the matching contract from everything as you said.

Reply:
Or you must use the full path here? Idk

Reply:
Checked it Harry, and it works well, cause it checks bytecodes, and not names.
We can add paths as well, but because we have different contracts with different paths, it'll make the code so ugly. So my vote would be not doing it cause it works well!
I will resolve this cm, if you think we need to add it please open the discussion again
Thanks!